module/advancement/types/scale-value.mjs

```diff
@@ -2,6 +2,195 @@ import Advancement from "../advancement.mjs";
+
+  /**
+   * This scale value prepared to be used in roll formulas.
+   * @type {string|null}
+   */
+  get prepared() { return this.value; }
+
+  /* -------------------------------------------- */
+
+  /**
+   * This scale value formatted for display.
+   * @type {string|null}
+   */
+  get formatted() { return this.prepared; }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Shortcut to the prepared value when used in roll formulas.
+   * @returns {string}
+   */
+  toString() {
+    return this.prepared;
+  }
+}
+
+
+export class ScaleValueTypeNumber extends ScaleValueType {
+
+  static get metadata() {
+    return foundry.utils.mergeObject(super.metadata, {
+      label: "DND5E.AdvancementScaleValueTypeNumber",
+      hint: "DND5E.AdvancementScaleValueTypeHintNumber",
+      isNumeric: true
+    });
+  }
+
+  /* -------------------------------------------- */
+
+  static converted(original) {
+    const value = Number(original.prepared);
+    if ( value.isNaN ) return null;
+    return { value };
+  }
+}
+
+
+export class ScaleValueTypeCR extends ScaleValueTypeNumber {
+
+  static get metadata() {
+    return foundry.utils.mergeObject(super.metadata, {
+      label: "DND5E.AdvancementScaleValueTypeCR",
+      hint: "DND5E.AdvancementScaleValueTypeHintCR"
+    });
+  }
+
+  /* -------------------------------------------- */
+
+  get formatted() {
+    switch (this.value) {
+      case 0.125: return "&frac18;";
+      case 0.25: return "&frac14;";
+      case 0.5: return "&frac12;";
+      default: return super.formatted;
+    }
+  }
+}
+
+
+export class ScaleValueTypeDice extends ScaleValueType {
+
+  static get metadata() {
+    return foundry.utils.mergeObject(super.metadata, {
+      label: "DND5E.AdvancementScaleValueTypeDice",
+      hint: "DND5E.AdvancementScaleValueTypeHintDice"
+    });
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * List of die faces that can be chosen.
+   * @type {number[]}
+   */
+  static FACES = [2, 3, 4, 6, 8, 10, 12, 20, 100];
+
+  /* -------------------------------------------- */
+
+  static converted(original) {
+    const split = (original.prepared ?? "").split("d");
+    if ( !split[1] ) return null;
+    const n = Number(split[0]);
+    const die = Number(split[1]);
+    if ( Number.isNaN(n) || Number.isNaN(die) ) return null;
+    return { n, die };
+  }
+
+  /* -------------------------------------------- */
+
+  get prepared() {
+    if ( !this.die ) return null;
+    return `${this.n ?? ""}${this.dice}`;
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * The number of dice to roll.
+   * @type {number}
+   */
+  get number() { return this.n ?? 0; }
+
+  /* -------------------------------------------- */
+
+  /**
+   * The die value to be rolled with the leading "d" (e.g. "d4").
+   * @type {string}
+   */
+  get dice() {
+    if ( !this.die ) return "";
+    return `d${this.die}`;
+  }
```

Comment:
Perhaps it might be better to return just the number?

Reply:
You can get just the number using `@scale.battle-master.superiority-dice.die`, which is admittedly a bit close to `dice`.

Reply:
I guess, I don't quite feel that this really needs to be a function. I would at least give it a more specific name, like `prefixedDie`.